lib/msf/core/web_services/module_search.rb

```diff
@@ -0,0 +1,72 @@
+module Msf::WebServices::ModuleSearch
+
+  def search_modules(opts)
+    raise ::ArgumentError, "At least one search parameter must be provided." if opts.except(:fields).empty?
+    search_results = []
+    metadata = Msf::Modules::Metadata::Cache.instance.get_metadata
+    params = parse_params(opts)
+    metadata.each do | module_metadata |
+      if Msf::Modules::Metadata::Cache.instance.matches(params, module_metadata)
```

Comment:
To keep the code paths consistent, I feel like the local and remote versions should still be using the same call to the actual search logic. I realize everything does funnel down to `Msf::Modules::Metadata::Search` eventually, but that's not immediately apparent. 

Right now it looks like `Msf::Modules::Metadata::Search` is the local equivalent of the search, and this just seems to be for the purpose of parsing the local param string first. I think that the caller of the actual search method (`is_match` in this case) should handle that prior to making the call. This way we can have a single method, with a single `opts` parameter passed in, that is consistent on either local or remote.

I know this isn't really an issue at this point because we don't currently have a use-case for a local msfconsole searching a remote module repo, but it will probably be happening at some point in the future.

Also, I'd love to hear @mkienow-r7 thoughts on this before any code changes are made.

Reply:
Reproducing my response here from chat last night. @jbarnett-r7 I think you raise a good point and I agree with the refactoring. It would increase readability of the remote search code, that is, `Msf::Modules::Metadata::Search` vs `Msf::Modules::Metadata::Cache`.